docs/specification/auth/oauth2.md

```diff
@@ -0,0 +1,285 @@
+---
+title: OAuth 2.0 Authentication
+type: docs
+weight: 51
+---
+
+{{< callout type="info" >}}
+**Protocol Revision**: {{< param protocolRevision >}}
+{{< /callout >}}
+
+The Model Context Protocol (MCP) supports OAuth 2.0 as a standardized authentication method, allowing secure authorization flows between clients and servers. Tokens will be securely communicated as part of the request body.
+
+## Protocol Flow
```

Comment:
Can you add some description to this diagram? I liked how you previously pointed out that authorization and resource server could be the same thing, for example.

Reply:
It might also be specifically helpful to fill this doc with some well-understood OAuth 2.0 keywords, so it's clear what corresponds to what from that spec.

Reply:
Changed diagram to match the RFC much more closely. Also added the explicit callout that auth+resource server can be the same thing.